src/main/SnakeAndLadders/Practice/Services/BoardGame.java

```diff
@@ -0,0 +1,48 @@
+package SnakeAndLadders.Practice.Services;
+
+import SnakeAndLadders.Practice.Model.Board;
+import SnakeAndLadders.Practice.Model.Dice;
+import SnakeAndLadders.Practice.Model.Players;
+
+public class BoardGame {
+    Board board;
+    private int noOfPlayers=0;
+    private int winningPosition = 100;
+    private Printer print;
+
+    public BoardGame(Board board) {
+        this.board = board;
+        this.print = new CMDPrinter();
+    }
+
+    public void startGame() {
+        noOfPlayers=board.getPlayers().size();
+        Dice dice = board.getDice();
+        int playerTurn = -1;
+        int move;
+        int nextPos;
+        while(true) {
+            playerTurn=(playerTurn+1)%noOfPlayers;
+            Players player = board.getPlayers().get(playerTurn);
```

Comment:
Avoid low-level detail. Should have instead exposed a function getPlayerWithId(id) in board.

Reply:
Read "Law of Demeter/Principle of Least Knowledge" here: https://workat.tech/machine-coding/tutorial/software-design-principles-abstraction-extensibility-cohesion-acafi2r32c78